src/App.tsx

```diff
@@ -3,33 +3,54 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string;
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  handleChange = (event: React.FormEvent<HTMLInputElement>) => {
+    this.setState({
+      query: event.currentTarget.value,
+    });
+  };
+
+  getVisibleMovies = () => {
+    return [...moviesFromServer].filter(movie => (
+      movie.title.toLocaleLowerCase().includes(this.state.query.toLocaleLowerCase())
+      || movie.description.toLocaleLowerCase().includes(this.state.query.toLocaleLowerCase())
+    ));
```

Comment:
Now, it looks too complex, try to optimize and simplify in such way:
```suggestion
    return [...moviesFromServer].filter({ title, description } => {
      	const lowerCaseQuery = this.state.query.toLowerCase();
      	
        return title.toLowerCase().includes(lowerCaseQuery)
          || description.toLowerCase().includes(lowerCaseQuery)
      });
```